gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineEnablePreviewFlag.java

```diff
@@ -29,26 +31,28 @@
 import org.gradle.api.tasks.testing.Test;
 import org.gradle.external.javadoc.CoreJavadocOptions;
 import org.gradle.process.CommandLineArgumentProvider;
-
 public final class BaselineEnablePreviewFlag implements Plugin<Project> {
 
     private static final String FLAG = "--enable-preview";
+    private static final String ENABLE_PREVIEW_PROPERTY = "enablePreview";
 
     @Override
     public void apply(Project project) {
         // The idea behind registering a single 'extra property' is that other plugins (like
-        // sls-packaging) can easily detect this and also also add the --enable-preview jvm arg
-        Provider<Boolean> enablePreview = project.provider(() -> {
-            JavaVersion jvmExecutingGradle = JavaVersion.current();
-            JavaPluginConvention javaConvention = project.getConvention().findPlugin(JavaPluginConvention.class);
-            if (javaConvention == null) {
-                return false;
+        // sls-packaging) can easily detect this and also also add the --enable-preview jvm arg.
+        project.getExtensions().getExtraProperties().set(ENABLE_PREVIEW_PROPERTY, project.provider(() -> {
```

Comment:
Yeah - I think stuff has evolved since I first made these commits... this is very much just so that I can get my local machine working, happy to rework to integrate with [that plugin!](https://github.com/palantir/gradle-baseline/blob/develop/gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineModuleJvmArgs.java).

I also wanted to figure out how explicit the opt-in to enable-preview _should_ be... I'd feel a bit weird about this magically being transitively applied to a project

Reply:
Perhaps a middle ground could be:
- the `--enable-preview` flag is automatically propagated to the launcher-static args used at runtime
- it is _not_ propagated to the compiler args of projects, so that each gradle project which wants to use fancy new features will need to explicitly opt-in.